plugins/buildingplan.cpp

```diff
@@ -670,13 +693,17 @@ DFhackCExport command_result plugin_onstatechange(color_ostream &out, state_chan
     return CR_OK;
 }
 
+static bool cycle_requested = false;
+
 #define DAY_TICKS 1200
 DFhackCExport command_result plugin_onupdate(color_ostream &)
 {
-    if (Maps::IsValid() && !World::ReadPauseState() && world->frame_counter % (DAY_TICKS/2) == 0)
+    if (Maps::IsValid() && !World::ReadPauseState()
```

Comment:
I've noticed that World::ReadPauseState() returns true when you hit space to pause the game and then go into query mode, but it returns false when you go directly into query mode and the game pauses as a side effect. Is there a good way to tell when the game is actually paused?

Reply:
I think any ui.main.mode value other than Default (or any viewscreen other than viewscreen_dwarfmodest) pauses the game. Unsure if this is wrapped in a library function anywhere.

Reply:
ok. I'll investigate this and improve pause detection in a future PR.